app/models/order_item.rb

```diff
@@ -45,17 +49,31 @@ def update_message(level, message)
   end
 
   def mark_completed(msg = nil, **opts)
-    mark_item(msg, :completed_at => DateTime.now, :state => "Completed", **opts)
+    return if state == "Completed"
+
+    mark_item(msg, :completed_at => Time.now.utc, :state => "Completed", **opts)
     Catalog::SubmitNextOrderItem.new(order_id).process
   end
 
   def mark_failed(msg = nil, **opts)
-    mark_item(msg, :completed_at => DateTime.now, :state => "Failed", :level => "error", **opts)
-    Catalog::SubmitNextOrderItem.new(order_id).process
+    return if state == "Failed"
+
+    mark_item(msg, :completed_at => Time.now.utc, :state => "Failed", :level => "error", **opts)
+    if order.state == 'Ordered'
+      # Order has been approved. One item fail but others may continue.
+      Catalog::SubmitNextOrderItem.new(order_id).process
+    else
+      # Order has not been approved. Fail all items.
+      order.order_items.each do |item|
+        item.mark_failed("Failed because its order is failing") unless item.finished?
```

Comment:
Just a bit of a clarification on the language here, maybe:
`"This order item has failed because the whole order has failed"`
or something like:
`"This order item has failed due to the entire order failing before it ran"`